client/src/main/java/org/eclipse/hono/client/impl/TelemetrySenderImpl.java

```diff
@@ -59,32 +60,14 @@ protected TelemetrySenderImpl(
         super(con, sender, tenantId, targetAddress);
     }
 
-    /**
-     * Gets the AMQP <em>target</em> address to use for uploading data to Hono's telemetry endpoint.
-     *
-     * @param tenantId The tenant to upload data for.
-     * @param deviceId The device to upload data for. If {@code null}, the target address can be used
-     *                 to upload data for arbitrary devices belonging to the tenant.
-     * @return The target address.
-     * @throws NullPointerException if tenant is {@code null}.
-     */
-    public static String getTargetAddress(final String tenantId, final String deviceId) {
-        final StringBuilder targetAddress = new StringBuilder(TelemetryConstants.TELEMETRY_ENDPOINT)
-                .append("/").append(Objects.requireNonNull(tenantId));
-        if (deviceId != null && deviceId.length() > 0) {
-            targetAddress.append("/").append(deviceId);
-        }
-        return targetAddress.toString();
-    }
-
     @Override
     public String getEndpoint() {
         return TelemetryConstants.TELEMETRY_ENDPOINT;
     }
 
     @Override
     protected String getTo(final String deviceId) {
-        return getTargetAddress(tenantId, deviceId);
+        return AddressHelper.getTargetAddress(TelemetryConstants.TELEMETRY_ENDPOINT, tenantId, deviceId, connection.getConfig().getAddressRewriteRule());
```

Comment:
how about keeping the *getTargetAddress* method and just change its implementation to delegate to `AddressHelper`. This would mean much fewer changes in the other code, wouldn't it?

Reply:
@sophokles73 It would still require changes everywhere, as we need to add/pass additional argument to this method. I was having it like that in the beginning but noted that the implementation was not consistent among classes, so I moved the logic to the helper method call. At that point I didn't see any value in these intermediary methods, so I removed them.

I can reintroduce them back if you want, I don't have a strong opinion on that.

Reply:
No worries, I see your point ...